pkg/segment/context/context.go

```diff
@@ -22,18 +27,45 @@ func NewContext(ctx context.Context) context.Context {
 
 // GetContextProperties retrieves all the values set in a given context
 func GetContextProperties(ctx context.Context) map[string]interface{} {
-	properties := propertiesFromContext(ctx)
-	if properties == nil {
+	cProperties := propertiesFromContext(ctx)
+	if cProperties == nil {
 		return make(map[string]interface{})
 	}
-	return properties.values()
+	return cProperties.values()
 }
 
-// SetComponentType sets componentType property for telemetry data when a new component is created
+// SetComponentType sets componentType property for telemetry data when a component is created/pushed
 func SetComponentType(ctx context.Context, value string) {
 	setContextProperty(ctx, "componentType", value)
 }
 
+// SetClusterType sets clusterType property for telemetry data when a component is pushed or a project is created/set
+func SetClusterType(ctx context.Context, client *occlient.Client) {
+	var value string
+	if client == nil {
+		value = "not-found"
+	} else {
+		// We are not checking ServerVersion to decide the cluster type because it does not always return the version,
+		// it sometimes fails to retrieve the data if user is using minishift or plain oc cluster
+		isOC, err := client.IsProjectSupported()
+		if err != nil {
+			klog.V(3).Info(errors.Wrap(err, "unable to detect project support"))
+			value = "not-found"
+		} else {
+			if isOC {
+				if isOC4, _ := client.GetKubeClient().IsCSVSupported(); isOC4 {
```

Comment:
Please check the error here as well

Reply:
I was wondering whether I should, because if it throws an error we still know that it is an openshift cluster. What value do you suggest we assign in case this throws an error?

Reply:
But in case of an connection error to the cluster, this might lead to false assumptions about the type of openshift cluster. 

>  What value do you suggest we assign in case this throws an error?

I would suggest not setting the property at all or defaulting to just `openshift`.

Reply:
I want to go with `openshift` but then it would mean adding a unit test for it which would mean mocking the `IsCSVSupported` function to return an error and all of this seems like a stretch to me.
If you still think that it's worth the effort, I'll go with it.

Reply:
OK, let's default to `openshift` when an error occurs. We can skip the unit test case for now.